diff --git a/_worker.js b/_worker2.js
index dec34bc..241b729 100644
--- a/_worker.js
+++ b/_worker2.js
@@ -27,11 +27,14 @@ let addresses = [];
 let addressesapi = [];
 let addressesnotls = [];
 let addressesnotlsapi = [];
 let addressescsv = [];
 let DLS = 8;
-let remarkIndex = 1;//CSV备注所在列偏移量
+let rename = 'Yutian81';
+let countrynum = 4;
+let citynum = 5;
+// let remarkIndex = 1;//CSV备注所在列偏移量
 let FileName = atob('ZWRnZXR1bm5lbA==');
 let BotToken;
 let ChatID;
 let proxyhosts = [];
 let proxyhostsURL = '';
@@ -120,11 +123,14 @@ export default {
                 if (env.ADDAPI) addressesapi = await 整理(env.ADDAPI);
                 if (env.ADDNOTLS) addressesnotls = await 整理(env.ADDNOTLS);
                 if (env.ADDNOTLSAPI) addressesnotlsapi = await 整理(env.ADDNOTLSAPI);
                 if (env.ADDCSV) addressescsv = await 整理(env.ADDCSV);
                 DLS = Number(env.DLS) || DLS;
-                remarkIndex = Number(env.CSVREMARK) || remarkIndex;
+                rename = env.RENAME || rename;
+                countrynum = parseInt(env.COUNTRYNUM || countrynum, 10);
+                citynum = parseInt(env.CITYNUM || citynum, 10);
+                // remarkIndex = Number(env.CSVREMARK) || remarkIndex;
                 BotToken = env.TGTOKEN || BotToken;
                 ChatID = env.TGID || ChatID;
                 FileName = env.SUBNAME || FileName;
                 subEmoji = env.SUBEMOJI || env.EMOJI || subEmoji;
                 if (subEmoji == '0') subEmoji = 'false';
@@ -1846,11 +1852,13 @@ async function 整理测速结果(tls) {
             const header = lines[0].split(',');
             const tlsIndex = header.indexOf('TLS');
 
             const ipAddressIndex = 0;// IP地址在 CSV 头部的位置
             const portIndex = 1;// 端口在 CSV 头部的位置
-            const dataCenterIndex = tlsIndex + remarkIndex; // 数据中心是 TLS 的后一个字段
+            // const dataCenterIndex = tlsIndex + remarkIndex; // 数据中心是 TLS 的后一个字段
+            const countryIndex = tlsIndex + countrynum;
+            const cityIndex = tlsIndex + citynum;
 
             if (tlsIndex === -1) {
                 console.error('CSV文件缺少必需的字段');
                 continue;
             }
@@ -1861,13 +1869,16 @@ async function 整理测速结果(tls) {
                 const speedIndex = columns.length - 1; // 最后一个字段
                 // 检查TLS是否为"TRUE"且速度大于DLS
                 if (columns[tlsIndex].toUpperCase() === tls && parseFloat(columns[speedIndex]) > DLS) {
                     const ipAddress = columns[ipAddressIndex];
                     const port = columns[portIndex];
-                    const dataCenter = columns[dataCenterIndex];
+                    //const dataCenter = columns[dataCenterIndex];
+                    //const formattedAddress = `${ipAddress}:${port}#${dataCenter}`;
+                    const country = columns[countryIndex];
+                    const city = columns[cityIndex];
+                    const formattedAddress = `${ipAddress}:${port}#${country} | ${city}@${rename}`;
 
-                    const formattedAddress = `${ipAddress}:${port}#${dataCenter}`;
                     newAddressescsv.push(formattedAddress);
                     if (csvUrl.includes('proxyip=true') && columns[tlsIndex].toUpperCase() == 'true' && !httpsPorts.includes(port)) {
                         // 如果URL带有'proxyip=true'，则将内容添加到proxyIPPool
                         proxyIPPool.push(`${ipAddress}:${port}`);
                     }
